bin/check-asg-instances-created.rb

```diff
@@ -57,41 +57,57 @@ class CheckAsgInstanceCreated < Sensu::Plugin::Check::CLI
          long:        '--critical_limit Critical Limit',
          description: 'Critical Limit for launching and terminated instances'
 
+  option :time_limit,
+         short:       '-t T',
+         long:        '--time Timelimit',
+         description: 'Time Limit for launching and terminated instances (default 1800s)'
+
   def asg
     @asg = Aws::AutoScaling::Client.new
   end
 
+  def ec2
+    @ec2 = Aws::EC2::Client.new
+  end
+
   def describe_activities(asg_group_name)
     asg.describe_scaling_activities(
       auto_scaling_group_name: asg_group_name.to_s
     )
   end
 
+  def describe_instance(instance_id)
+    ec2.describe_instances(
+      instance_ids: [instance_id]
+    )
+  end
+
   def run
-    warning = 3
-    critical = 4
-    result_launched = ''
+    warning = 2
+    critical = 3
+    time = 1800
     result_terminated = ''
-    instance_launching = 0
     instance_terminating = 0
     time_now = Time.now
     time_utc_offset = time_now - time_now.utc_offset
 
-    if !config[:warning_limit].nil?
+    unless config[:warning_limit].nil?
       warning = config[:warning_limit]
-    elsif !config[:critical_limit].nil?
+    end
+    unless config[:critical_limit].nil?
       critical = config[:critical_limit]
     end
+    unless config[:time_limit].nil?
+      time = config[:time_limit].to_i
+    end
 
     if config[:asg_group_name].nil?
       asg.describe_auto_scaling_groups.auto_scaling_groups.each do |asg_group|
         describe_activities(asg_group.auto_scaling_group_name).each do |activities|
           activities.activities.each do |activity|
-            if Time.parse(activity.start_time.inspect) > (time_utc_offset - 3600)
-              if activity.description.include? 'Launching'
-                instance_launching += 1
-                result_launched = " #{instance_launching} Instances Launching in AutoScalingGroup #{asg_group.auto_scaling_group_name}"
-              elsif activity.description.include? 'Terminating'
+            if Time.parse(activity.start_time.inspect) > (time_utc_offset - time) && activity.description.include?('Terminating')
+              id = activity.description.gsub!(/^.*: ([a-z0-9-]+)$/, '\1')
+              if Time.parse(describe_instance(id).reservations[0].instances[0].launch_time.inspect) > (time_utc_offset - time)
```

Comment:
Could you give me an example object? Just want to ensure that `.reservations[0].instances[0]` makes sense and is not brittle.